### PR TITLE
[ALLUXIO-3293] Fix retry policy being invalid when reading local block from a crashed local worker

### PR DESCRIPTION
**Pull request by snodawn**
https://alluxio.atlassian.net/browse/ALLUXIO-3293

@apc999 @aaudiber 
I'm sorry to miss closing PR in https://github.com/Alluxio/alluxio/pull/7785, and try hard to re-open it, but no effect. So I create a new one here.

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/FileInStream.java [hunk 1]
```diff
@@ -163,8 +165,10 @@ public int read(byte[] b, int off, int len) throws IOException {
         lastException = null;
       } catch (UnavailableException | ConnectException | DeadlineExceededException e) {
         lastException = e;
-        handleRetryableException(mBlockInStream, e);
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/FileInStream.java]: put line 156 `updateStream();` into this `try` block too?

**Comment by apc999** [comment]
@apc999 cherry-pick this to 1.8 branch after it is merged.

File: core/client/fs/src/main/java/alluxio/client/file/FileInStream.java (hunk 1 above)
Comment (snodawn) [reply core/client/fs/src/main/java/alluxio/client/file/FileInStream.java]: I have put a new submit to fix it.

**Comment by snodawn** [comment]
@apc999 @aaudiber I have fixed it, please check again, thanks.

**Comment by apc999** [comment]
@snodawn thanks for the patch! I will also cherry-pick your fix to branch 1.8

**Comment by snodawn** [comment]
@apc999 @aaudiber thank you for reviewing it!

**Pull request merged**